nf_core/download.py

```diff
@@ -1664,9 +1664,7 @@ def tidy_tags_and_branches(self):
                     else:
                         # desired revisions may contain arbitrary branch names that do not correspond to valid sematic versioning patterns.
                         valid_versions = [
-                            version_parser(v)
-                            for v in desired_revisions
-                            if re.match(r"\d+\.\d+(?:\.\d+)*(?:[\w\-_])*", v)
+                            str(Version(v)) for v in desired_revisions if re.match(r"\d+\.\d+(?:\.\d+)*(?:[\w\-_])*", v)
```

Comment:
A separate issue is the conversion of the `Version` instance  to a string. It will lead to wrong orderings as demonstrated below with a version number which had a release candidate.

```
>>> sorted(str(Version(v)) for v in ['1.23.4', '1.23.4-1', '1.23.5', '1.23.5rc'])[-1]
'1.23.5rc0'
>>> sorted(Version(v) for v in ['1.23.4', '1.23.4-1', '1.23.5', '1.23.5rc'])[-1] 
<Version('1.23.5')>
```